sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -268,18 +268,27 @@ private[parquet] class ParquetRowConverter(
         }
 
       // For INT32 backed decimals
-      case t: DecimalType if parquetType.asPrimitiveType().getPrimitiveTypeName == INT32 =>
-        new ParquetIntDictionaryAwareDecimalConverter(t.precision, t.scale, updater)
+      case _: DecimalType if parquetType.asPrimitiveType().getPrimitiveTypeName == INT32 =>
+        val metadata = parquetType.asPrimitiveType().getDecimalMetadata
+        val precision = metadata.getPrecision()
+        val scale = metadata.getScale()
+        new ParquetIntDictionaryAwareDecimalConverter(precision, scale, updater)
```

Comment:
same question as the avro PR: how do we handle the precision/scale inconsistency between the decimal value and the catalyst decimal type?

Reply:
1. For object handling, after creating decimal object, the upper layer uses the object's precision and scale to handle the object's value. Do you have some usual-suspect code?
2. For the user-facing side, this doesn't change the user-provided schema. So, there is no change from user's perspective.